docs/source/user_guide/writing-extractors.rst

```diff
@@ -0,0 +1,155 @@
+For example, extractors in packages developed under the datalad organisation could use ``<extractor_name>.datalad.org``.
+Example generation with Python: ``uuid.uuid5(uuid.NAMESPACE_DNS, "bids_dataset.datalad.org")``.
+
+Example::
+
+  def get_id(self) -> UUID:
+      return UUID("d1e3f728-7868-4f33-a4a5-0f9f57dce342")
+
+``get_version()``
+-----------------
+
+This function should return a version string representing the extractor's version.
+
+The extractor version is meant to be included with the extracted metadata.
+It might be a version number, or a reference to a particular schema.
+While there are no requirements on the versioning scheme, it should be chosen carefully, and rules for version updating need to be considered.
+For example, when using `semantic versioning <https://semver.org/>`_ style, one might decide to guarantee that attribute type and name will never change within a major release, while new attributes may be added with a minor release.
+
+Example::
+
+  def get_version(self) -> str:
+      return "0.0.1"
+
+``get_data_output_category()``
+------------------------------
+
+This function should return a ``DataOutputCategory`` object, which tells MetaLad what kind of (meta)data it is dealing with.
+
+.. TODO:: What categories are available, what are the consequences of declaring one?
+
+Example::
+  
+  def get_data_output_category(self) -> DataOutputCategory:
+      return DataOutputCategory.IMMEDIATE
+
+``get_required_content()``
+--------------------------
+
+This function is used in dataset-extractors only.
+It will be called by MetaLad prior to metadata extraction.
+Its purpose is to make sure that content required for metadata extraction is present
+(relevant, for example, if some of files to be inspected may be annexed).
+The function should return ``True`` if it has obtained the required content, or confirmed its presence.
+If it returns ``False``, metadata extraction will not proceed.
+
+This function can be a place to call ``dataset.get()``.
+It is advisable to disable result rendering (``result_renderer="disabled"``), because during metadata extraction, users will typically want to redirect standard output to a file or another command.
+
+Example::
+
+  def get_required_content(self) -> bool:
+      result = self.dataset.get("CITATION.cff", result_renderer="disabled")
+      return result[0]["status"] in ("ok", "notneeded")
+
+``is_content_required()``
+-------------------------
+
+This function is used in file-extractors only.
+It is a file-extractor counterpart to ``get_required_content``.
+Its purpose is to tell MetaLad if the file content is required or not
+(relevant for annexed files - extraction may depend on file content, or require only annex key).
+If the function returns ``True``, MetaLad will get the file content.
+If it returns ``False``, the get operation will not be performed.
```

Comment:
I didn't work with file-level extractors, that's my guess - is that true?

Also, might be good to come up with an example.

Reply:
Confirmed, this is how it works.